internal/pkg/types/signed_message.go

```diff
@@ -83,12 +84,31 @@ func (smsg *SignedMessage) Cid() (cid.Cid, error) {
 // ToNode converts the SignedMessage to an IPLD node.
 func (smsg *SignedMessage) ToNode() (ipld.Node, error) {
 	// Use 32 byte / 256 bit digest.
-	obj, err := cbor.WrapObject(smsg, DefaultHashFunction, -1)
+	mhType := uint64(mh.BLAKE2B_MIN + 31)
+	mhLen := -1
+
+	data, err := encoding.Encode(smsg)
+	if err != nil {
+		return nil, err
+	}
+
+	hash, err := mh.Sum(data, mhType, mhLen)
+	if err != nil {
+		return nil, err
+	}
+	c := cid.NewCidV1(cid.DagCBOR, hash)
+
+	blk, err := blocks.NewBlockWithCid(data, c)
+	if err != nil {
+		return nil, err
+	}
+	obj, err := cbor.DecodeBlock(blk)
```

Comment:
Actually decoding it seems pretty roundabout, considering we have the data at line 90.

Reply:
The names aren't great but this is the straightforward path.  It is the function that maps from a cbor binary/cided blocks.block to an ipld node.